pkg/logger/worker.go

```diff
@@ -75,85 +84,145 @@ func NewWorker(id int, workerQueue chan chan LogRequest, logger *zap.SugaredLogg
-		cloudevents.WithTimeNow(),
-	)
+func (w *Worker) sendBatchCloudEvent(logReqs []LogRequest) error {
+	events := make([]cloudevents.Event, 0, len(logReqs))
+
+	for _, lr := range logReqs {
+		event, err := w.prepareEvent(lr)
+		if err != nil {
+			return err
+		}
+		events = append(events, event)
+	}
+
+	c, err := w.buildClient(logReqs[0].Url, logReqs[0].CertName, logReqs[0].TlsSkipVerify)
 	if err != nil {
-		return fmt.Errorf("while creating new cloudevents client: %w", err)
+		return fmt.Errorf("while creating client: %w", err)
 	}
-	event := cloudevents.NewEvent(cloudevents.VersionV1)
-	event.SetID(logReq.Id)
-	event.SetType(logReq.ReqType)
 
-	event.SetExtension(InferenceServiceAttr, logReq.InferenceService)
-	event.SetExtension(NamespaceAttr, logReq.Namespace)
-	event.SetExtension(ComponentAttr, logReq.Component)
-	event.SetExtension(EndpointAttr, logReq.Endpoint)
+	httpReq, err := cloudevents.NewHTTPRequestFromEvents(w.CeCtx, logReqs[0].Url.String(), events)
+	if err != nil {
+		return fmt.Errorf("while creating http request: %w", err)
+	}
+	// Send event
+	res, err := c.Do(httpReq)
+	if err != nil {
+		return fmt.Errorf("while sending event: %w", err)
+	}
+	// Close the body
+	defer res.Body.Close()
 
-	encodedMetadata, err := json.Marshal(logReq.Metadata)
+	if res.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(res.Body)
+		if err != nil {
+			w.Log.Warnf("unable to read body of response: %w", err)
+		}
+		err = fmt.Errorf(string(body))
+	}
+	w.Log.Infof("Sent with status code %d, error: %v", res.StatusCode, err)
+	return nil
+
+}
+
+func (w *Worker) sendCloudEvent(logReq LogRequest) error {
+	// Create a format event
+	event, err := w.prepareEvent(logReq)
 	if err != nil {
-		return fmt.Errorf("could not encode metadata as json: %w", err)
+		return err
 	}
-	event.SetExtension(MetadataAttr, string(encodedMetadata))
 
-	event.SetSource(logReq.SourceUri.String())
-	if err := event.SetData(logReq.ContentType, *logReq.Bytes); err != nil {
-		return fmt.Errorf("while setting cloudevents data: %w", err)
+	// Create the client
+	c, err := w.buildClient(logReq.Url, logReq.CertName, logReq.TlsSkipVerify)
+	if err != nil {
+		return fmt.Errorf("while creating client: %w", err)
 	}
 
-	res := c.Send(w.CeCtx, event)
-	if cloudevents.IsUndelivered(res) {
-		return fmt.Errorf("while sending event: %w", res)
-	} else {
-		var httpResult *cehttp.Result
-		if cloudevents.ResultAs(res, &httpResult) {
-			var err error
-			if httpResult.StatusCode != http.StatusOK {
-				err = fmt.Errorf(httpResult.Format, httpResult.Args...)
-			}
-			w.Log.Infof("Sent with status code %d, error: %v", httpResult.StatusCode, err)
-		} else {
-			w.Log.Infof("Send did not return an HTTP response: %s", res)
+	// Generate Http request from event
+	httpReq, err := cloudevents.NewHTTPRequestFromEvent(w.CeCtx, logReq.Url.String(), event)
+	if err != nil {
+		return fmt.Errorf("while creating http request: %w", err)
+	}
+
+	// Send event
+	res, err := c.Do(httpReq)
```

Comment:
Do we have timeout for the http request

Reply:
The default behavior for the client is no timeout. I just checked if we have a timeout in the current way we send cloud events and I don't believe that has a timeout either. The http implementation of the send protocol which the CloudEvent client uses doesn't seem to set a default timeout. 

I put a timeout of 10 seconds, since that seems like a sensible default. I could see a case for increasing that to 30-60 seconds to be closer to the current behavior of no timeout. @yuzisun what are you thoughts?